### ISSUE TITLE
Aggregation in SemanticLoss

### ISSUE THREAD

**Issue by LedaguenelArthur** [lucadiliello/semantic-loss-pytorch#5]
Hi @lucadiliello,

I might be wrong but it seems like the SemanticLoss is doing the mean agregation on the batch before passing through the log, right ?
```
wmc = torch.mean(wmc_per_sample)
loss = -torch.log(wmc)
```
It seems to me it makes more sense to do the mean agregation after the log to respect the probabilistic semantics and match what is done for classical cross-entropy. Do you know what is used in the paper's version ?

Tell me what you think,
Best regards,
Arthur

**Comment by lucadiliello** [comment]
Hi @LedaguenelArthur, please have a look at this issue #1

**Comment by LedaguenelArthur** [comment]
Thanks a lot, and sorry for the inconvenience.
